internal/kldevents/eventstream.go

```diff
@@ -90,6 +90,9 @@ type eventStream struct {
 	batchCount        uint64
 	initialRetryDelay time.Duration
 	backoffFactor     float64
+	updateInProgress  bool
+	updateInterrupt   chan struct{}
```

Comment:
think it's best to be explicit on the type here

Reply:
Using a zero-sized `struct{}` to implement an alternative to `context.Context` is a recommendation I found in multiple blogs. ex:

https://www.ardanlabs.com/blog/2017/10/the-behavior-of-channels.html

`If you choose to use your own channel for cancellation, rather than the context package, your channel should be of type chan struct{}. It is the zero-space, idiomatic way to indicate a channel used only for signalling.`

Reply:
Makes sense - so it's defining an empty struct (rather than generic)

Reply:
i'll add a comment to make that point, i agree the intent isn't clear as-is